packages/msw-addon/src/initializeWorker.js

```diff
@@ -0,0 +1,66 @@
+import { setupWorker } from 'msw'
+import { setupServer } from 'msw/node'
+
+export class Worker {
+  constructor() {
+    this.worker = setupWorker()
+  }
+
+  start(options) {
+    this.worker.start(options)
+  }
+
+  use(...handlers) {
+    this.worker.use(...handlers)
+  }
+
+  resetHandlers(...nextHandlers) {
+    this.worker.resetHandlers(...nextHandlers)
+  }
+
+  restoreHandlers() {
+    this.worker.restoreHandlers()
+  }
+
+  printHandlers() {
+    this.worker.printHandlers()
+  }
+
+  stop() {
+    this.worker.stop()
+  }
+}
+
+export class Server {
+  constructor() {
+    this.server = setupServer()
+  }
+
+  start(options) {
+    this.server.listen(options)
+  }
+
+  use(...handlers) {
+    this.server.use(...handlers)
+  }
+
+  resetHandlers(...nextHandlers) {
+    this.server.resetHandlers(...nextHandlers)
+  }
+
+  restoreHandlers() {
+    this.server.restoreHandlers()
+  }
+
+  printHandlers() {
+    this.server.printHandlers()
+  }
+
+  stop() {
+    this.server.close()
+  }
+}
+
+const IS_BROWSER = typeof global.process === 'undefined'
+
+export const initializeWorkerInstance = () => (IS_BROWSER ? new Worker() : new Server())
```

Comment:
Once again, this brings in confusing terminology: we are not dealing with workers in a Node.js context. 

We may try to have a conditional setup directly in the `initializeWorker` function to being with:

```js
// packages/msw-addon/src/mswDecorator.js
const IS_BROWSER = typeof global.process === 'undefined'

export function initialize(options) {
  let api

  if (IS_BROWSER) {
    const { setupWorker } = require('msw')

    const worker = setupWorker()
    worker.start(options)
    api = worker
  } else {
    const { setupServer } = require('msw/node')

    const server = setupServer()
    server.listen(options)
    api = server
  }

  return api
}
```

Reply:
Thanks for your opinion! I was trying to add the fallback mechanism without changing the perception for the end user. I'm not super great with naming, so this comment helps! The only downside to this in comparison to the wrapper is that the return type of this function will be something like `BrowserAPI | ServerAPI`, so if we have ts types, the user might always have to typecast stuff if they want type safety

Reply:
I will take a stab at this, and I guess we figure the typing out when this is migrated to typescript?

Reply:
Same applies for the type of `Options`

Reply:
Also, do you think the names should be updated? `initializeWorker` -> `initialize`, `getWorker` -> `getApi`, something like that? the first one would be a breaking change